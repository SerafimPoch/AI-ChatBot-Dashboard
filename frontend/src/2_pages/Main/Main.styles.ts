import styled from "@emotion/styled";

export const MainContainer = styled.div`
  margin: 20px 64px 24px 20px;
  padding: 0 64px;
`;

export const MainContentWrapper = styled.div`
  display: grid;
  grid-template-columns: 1fr 0.5fr 1fr;
`;
