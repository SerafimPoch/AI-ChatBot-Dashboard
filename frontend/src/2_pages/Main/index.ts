export { default as MainPage } from "./view";
export { default as RootStyleRegistry } from "./config/emotion";
