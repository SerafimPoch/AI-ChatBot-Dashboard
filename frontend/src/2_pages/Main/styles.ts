import { useThemeContext } from "@/src/6_shared/contexts/ThemeContext";
import { LAPTOP_MQ, TABLET_MQ } from "@/src/6_shared/media-queries";
import styled from "@emotion/styled";

export const MainContainer = styled.main`
  ${() => {
    const themeColors = useThemeContext();

    return `
      width: 100vw;
      padding: 20px 64px 20px 20px;
      background: ${themeColors?.coreBackground};
      display: flex;
      justify-content: center;
      gap: 64px;
      align-items: stretch;

      @media(max-width:${LAPTOP_MQ}px) {
        gap: 40px;
      }

      @media(max-width:${TABLET_MQ}px) {
        gap: 20px;
        padding: 20px;
      }
    `;
  }}
`;

export const Container = styled.div`
  width: 100%;
  max-width: 1440px;
`;

export const MainContentWrapper = styled.div`
  display: grid;
  grid-template-columns: minmax(700px, 1.5fr) minmax(500px, 1fr);
  grid-row-gap: 8px;
  grid-column-gap: 8px;

  @media (max-width: ${LAPTOP_MQ}px) {
    grid-template-columns: 1fr 1fr;
  }

  @media (max-width: ${TABLET_MQ}px) {
    grid-template-columns: 1fr;
  }
`;
