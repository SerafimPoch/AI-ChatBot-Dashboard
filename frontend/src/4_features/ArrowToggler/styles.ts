import { useThemeContext } from "@/src/6_shared/contexts/ThemeContext";
import styled from "@emotion/styled";

export const ArrowTogglerContainer = styled.div`
  ${() => {
    const themeColors = useThemeContext();

    return `
      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: center;
      gap: 32px;
      align-self: stretch;
      border-radius: 12px;
      background: ${themeColors?.grey};
      width: 56px;
    `;
  }}
`;
