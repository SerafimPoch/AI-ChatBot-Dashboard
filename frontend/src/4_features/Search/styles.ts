import { useThemeContext } from "@/src/6_shared/contexts/ThemeContext";
import styled from "@emotion/styled";

export const SearchContainer = styled.div`
  position: relative;
  display: flex;
  align-items: center;
  gap: 8px;
`;

export const SearchInput = styled.input`
  ${() => {
    const themeColors = useThemeContext();

    return `
      background: transparent;
      border: 1px solid #5d74f1;
      outline: none;
      border-radius: 6px;
      padding: 0 5px;
      color: ${themeColors?.textPrimary};
    `;
  }}
`;
