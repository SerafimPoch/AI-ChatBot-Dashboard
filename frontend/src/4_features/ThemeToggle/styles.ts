import styled from "@emotion/styled";

interface IThemeItem {
  active: boolean;
}

export const ThemeToggleContainer = styled.div`
  display: flex;
  align-items: center;
  border-radius: 8px;
  border: 2px solid #f2f2f2;
  width: 100%;
  height: 40px;
`;

export const ThemeItem = styled.button<IThemeItem>`
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 6px;
  width: 50%;
  height: 32px;
  background: ${({ active }) => (active ? "#f2f2f2" : "transparent")};
  color: ${({ active }) =>
    active ? "var(--text-primary)" : "var(--text-secondary)"};
  margin: 0 4px;
  transition: background-color 0.3s ease, color 0.3s ease;
`;
