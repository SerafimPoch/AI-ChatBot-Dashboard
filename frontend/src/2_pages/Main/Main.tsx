"use client";
import { Header } from "@/src/3_widgets/Header";
import { MainContainer } from "./Main.styles";
import { ProfileModal } from "@/src/3_widgets/ProfileModal";
import { useStore } from "@nanostores/react";
import { $isProfileModalOpen } from "@/src/6_shared/store";

export default function MainPage() {
  const isProfileModalOpen = useStore($isProfileModalOpen);

  return (
    <MainContainer>
      {isProfileModalOpen && <ProfileModal />}
      <Header />
    </MainContainer>
  );
}
