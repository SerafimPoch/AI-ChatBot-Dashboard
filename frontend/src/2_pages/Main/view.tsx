"use client";
import { useStore } from "@nanostores/react";
import dynamic from "next/dynamic";
import { Header } from "@/src/3_widgets/Header";
import { ProfileModal } from "@/src/3_widgets/ProfileModal";
import { Audiences } from "@/src/2_pages/Main/ui/Audiences";
import { $isProfileModalOpen } from "@/src/6_shared/store";
import { SideBar } from "@/src/3_widgets/SideBar";
import { Statistic } from "./ui/Statistic";
import { Earnings } from "./ui/Earnings";
import { Container, MainContainer, MainContentWrapper } from "./styles";

const Membership = dynamic(() =>
  import("@/src/5_entities/Membership").then((mod) => mod.Membership)
);

export default function MainPage() {
  const isProfileModalOpen = useStore($isProfileModalOpen);

  return (
    <MainContainer>
      {isProfileModalOpen && <ProfileModal />}
      <SideBar />
      <Container>
        <Header />
        <MainContentWrapper>
          <Statistic />
          <Membership />
          <Audiences />
          <Earnings />
        </MainContentWrapper>
      </Container>
    </MainContainer>
  );
}
