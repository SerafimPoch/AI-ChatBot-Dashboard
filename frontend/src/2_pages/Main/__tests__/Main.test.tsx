import React from "react";
import { render } from "@testing-library/react";
import { MainContentWrapper } from "../Main.styles";

describe("MainContentWrapper", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<MainContentWrapper />);
    expect(asFragment()).toMatchSnapshot();
  });
});
