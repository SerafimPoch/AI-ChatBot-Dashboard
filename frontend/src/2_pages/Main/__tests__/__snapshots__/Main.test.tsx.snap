// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContentWrapper renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: grid;
  grid-template-columns: minmax(700px, 1.5fr) minmax(500px, 1fr);
  grid-row-gap: 8px;
  grid-column-gap: 8px;
}

@media (max-width: 1550px) {
  .emotion-0 {
    grid-template-columns: 1fr 1fr;
  }
}

@media (max-width: 1240px) {
  .emotion-0 {
    grid-template-columns: 1fr;
  }
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;
